### PR TITLE
Issue 61 - Update Symfony 5.4

### PR DESCRIPTION
**Pull request by BowlOfSoup**
* Removed support for Symfony ~4.4
* Support for Symfony 5.4 (PHP 7.2 up until 8.2)
* Lots of Rector and Stan changes

### CODE REVIEW

File: tests/Service/Extractor/MethodExtractorTest.php [hunk 1]
```diff
@@ -10,15 +10,15 @@
 
 class MethodExtractorTest extends TestCase
 {
-    /** @var \BowlOfSoup\NormalizerBundle\Service\Extractor\MethodExtractor|\PHPUnit\Framework\MockObject\MockObject */
+    /** @var \PHPUnit\Framework\MockObject\MockObject|(\BowlOfSoup\NormalizerBundle\Service\Extractor\MethodExtractor&\PHPUnit\Framework\MockObject\MockObject) */
```
Comment (gabriel-255) [inline tests/Service/Extractor/MethodExtractorTest.php:13]: Just a tip: if you want to have a slightly easier time using PHPUnit mocks next to PHPStan, there's a PHPStan extension that helps smooth out some of the idiomatic stuff (like mock union/intersection types), similar to the mockery extension that I already added last week (I didn't look, I thought the project was using Mockery).

https://github.com/phpstan/phpstan-phpunit#:~:text=Interprets%20Foo%7CPHPUnit_Framework_MockObject_MockObject%20in%20phpDoc%20so%20that%20it%20results%20in%20an%20intersection%20type%20instead%20of%20a%20union%20type.

File: composer.json [hunk 2]
```diff
@@ -15,18 +15,17 @@
...
+    "doctrine/common": "~3.0",
     "doctrine/collections": "1.*",
-    "php-coveralls/php-coveralls": "^2.0",
-    "phpunit/phpunit": "^8.5",
-    "friendsofphp/php-cs-fixer": "^2.10",
+    "phpunit/phpunit": "^8.0",
+    "friendsofphp/php-cs-fixer": "^3.0",
     "rector/rector": "^0.18.6",
     "phpstan/phpstan": "^1.10",
     "phpstan/phpstan-mockery": "^1.1",
```
Comment (gabriel-255) [inline composer.json]: Since the unit tests are not (currently) using Mockery, you can remove this.

File: phpstan.neon.dist
Comment (gabriel-255) [inline phpstan.neon.dist:1]: Since the unit tests are not (currently) using Mockery, you can remove the mockery extension.

**Review by gabriel-255 — APPROVED** [review]

File: composer.json (hunk 2 above)
Comment (BowlOfSoup) [reply composer.json]: Maybe we will in the future. Will remove it for now.

**Pull request merged**